### PR TITLE
Add `OPTCausalLM` and preprocessors

### PR DESCRIPTION
**Pull request by chenmoneygithub**
One fact we need to check is when I played with OPT generation, the text is chaotic. I am suspecting it is due to the converted weights and not specifically related to the code in this PR.

### CODE REVIEW

**Comment by ADITYADAS1999** [comment]
Related  #921

File: keras_nlp/models/opt/opt_causal_lm.py [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+        )
+
+    def generate(
+        self,
+        inputs,
+        max_length=None,
+    ):
+        """Generate text given prompt `inputs`.
+
+        This method generates text based on given `inputs`. The sampling method
```
Comment (mattdangerw) [inline keras_nlp/models/opt/opt_causal_lm.py:364]: It looks like there are some minor copy edits here to the GPT version. Can you mirror these changes for GPT?

File: keras_nlp/models/opt/opt_causal_lm.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+            token_ids, padding_mask = batch["token_ids"], batch["padding_mask"]
+            # If `preprocessor` is attached, we can stop after `end_token_id``.
+            end_token_id = None
+            if self.preprocessor is not None:
+                end_token_id = self.preprocessor.tokenizer.end_token_id
+            # Run the compiled generate function.
+            output = generate_function(token_ids, padding_mask, end_token_id)
+
+            if self.preprocessor is not None:
+                # Truncate to ragged by removing tokens after the first
```
Comment (mattdangerw) [inline keras_nlp/models/opt/opt_causal_lm.py:433]: Same here.

**Review by mattdangerw — APPROVED** [review]
This looks good! Have we tried out a training/generation run with the tiniest checkpoint to make sure things work ok?

**Pull request merged**